googlecloud-plugins/src/main/java/co/cask/hydrator/plugin/source/BigQuerySource.java

```diff
@@ -0,0 +1,222 @@
+  private static final String MRBQ_JSON_KEY = "mapred.bq.auth.service.account.json.keyfile";
+  private static final String FSGS_JSON_KEY = "google.cloud.auth.service.account.json.keyfile";
+  private static final Gson GSON = new Gson();
+  private static final Type MAP_STRING_STRING_TYPE = new TypeToken<Map<String, String>>() { }.getType();
+  private Map<String, String> outputSchemaMapping = new HashMap<>();
+  private  Schema outputSchema;
+  public BigQuerySource(BQSourceConfig config) {
+    super(new ReferencePluginConfig(config.referenceName));
+    this.sourceConfig = config;
+  }
+
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    init();
+  }
+
+  private void init() {
+    String[] schemaList = sourceConfig.outputSchema.split(",");
+    for (String schema : schemaList) {
+      String[] columns = schema.split(":");
+      outputSchemaMapping.put(columns[0], columns[1]);
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+    init();
+    getOutputSchema();
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(outputSchema);
+  }
+
+  @Override
+  public void prepareRun(BatchSourceContext context) throws IOException, GeneralSecurityException,
+    ClassNotFoundException, InterruptedException {
+    Job job = JobUtils.createInstance();
+    Configuration conf = job.getConfiguration();
+    conf.set(MRBQ_JSON_KEY, sourceConfig.jsonFilePath);
+    conf.set(FSGS_JSON_KEY, sourceConfig.jsonFilePath);
+    conf.set(BigQueryConfiguration.PROJECT_ID_KEY, sourceConfig.projectId);
+    if (sourceConfig.importQuery != null) {
+      conf.set(BigQueryConfiguration.INPUT_QUERY_KEY, sourceConfig.importQuery);
+    }
+      conf.set(BigQueryConfiguration.TEMP_GCS_PATH_KEY, sourceConfig.tmpBucketPath);
+    conf.set("fs.gs.project.id", sourceConfig.projectId);
+    BigQueryConfiguration.configureBigQueryInput(conf, sourceConfig.fullyQualifiedInputTableId);
+    job.setOutputKeyClass(LongWritable.class);
+    job.setOutputValueClass(Text.class);
+    context.setInput(Input.of(sourceConfig.referenceName,
+                              new SourceInputFormatProvider(JsonTextBigQueryInputFormat.class, conf)));
+  }
+
+  @Override
+  public void transform(KeyValue<LongWritable, Text> input, Emitter<StructuredRecord> emitter) throws IOException {
+    getOutputSchema();
+    StructuredRecord record = jsonTransform(input.getValue());
+    emitter.emit(record);
+
+  }
+
+  private StructuredRecord jsonTransform(Text jsonText) {
+    Map<String, String> map;
+    map = GSON.fromJson(jsonText.toString(), MAP_STRING_STRING_TYPE);
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+    for (Schema.Field field : outputSchema.getFields()) {
+      String fieldName = field.getName();
+      Schema schema = field.getSchema();
+      if (schema.getType() == Schema.Type.INT) {
```

Comment:
Also, do we have to worry about nullable output fields? If the field is for a nullable int for example then `field.getSchema()` will be a `UNION` of null and `INT`.